support language code in ElevenLabs TTS

cch41 opened the pull request:
While only supported for turbo v2.5, specifying language can be a helpful constraint. We currently overwrite your eleven labs tts to do this, upstreaming it here.

bcherry commented on livekit-plugins/livekit-plugins-elevenlabs/livekit/plugins/elevenlabs/tts.py:
```diff
@@ -141,6 +144,9 @@ def __init__(
             raise ValueError("ELEVEN_API_KEY must be set")
+        
+        if language_code is not None and model != "eleven_turbo_v2_5":
+            raise ValueError("language_code is only supported for the 'eleven_turbo_v2_5' model")
```
It would be better to leave this to the ElevenLabs API. What is its behavior right now if you send language_code on a different model? Is it ignored, or is it a 4xx error?

cch41 replied:
It throws a 400 error. Want me to just log a warning message instead? Also, should I disclude language_code if model != eleven_turbo_v2_5 in the api call then?

cch41 replied:
set it to None in this case and logged a warning

bcherry commented:
@theomonnom this seems reasonable and pretty much identical to the way this is supported for the Cartesia plugin already https://github.com/livekit/agents/blob/main/livekit-plugins/livekit-plugins-cartesia/livekit/plugins/cartesia/tts.py#L77

theomonnom commented on livekit-plugins/livekit-plugins-elevenlabs/livekit/plugins/elevenlabs/tts.py:
```diff
@@ -80,6 +80,7 @@ class _TTSOptions:
     api_key: str
     voice: Voice
     model: TTSModels | str
+    language_code: str | None
```
I think we should rename language_code to language for consistency

cch41 replied:
renamed

bcherry replied:
fwiw i disagree - language_code is the underlying name in the elevenlabs API

theomonnom reviewed (approved):
otherwise lgtm

cch41 commented:
Think the check fails are unrelated

cch41 commented:
@theomonnom mind re-running the tests so this can be merged?

theomonnom commented on livekit-plugins/livekit-plugins-elevenlabs/livekit/plugins/elevenlabs/tts.py:
```diff
@@ -141,6 +144,10 @@ def __init__(
         api_key = api_key or os.environ.get("ELEVEN_API_KEY")
         if not api_key:
             raise ValueError("ELEVEN_API_KEY must be set")
+        
+        if language is not None and model != "eleven_turbo_v2_5":
+            language = None
+            logger.warning("Excluding 'language' as it is only supported for the 'eleven_turbo_v2_5' model")
```
Let's do nothing by default? I think 11labs will throw an error if we use a different language no?
Otherwise lgtm

cch41 replied:
Yea it will throw an error, will remove this

theomonnom commented:
Can you run `ruff format` and `ruff check --fix`?

cch41 commented:
>Can you run ruff format and ruff check --fix?
<img width="243" alt="Screenshot 2024-11-01 at 1 45 17 PM" src="https://github.com/user-attachments/assets/7f8db8b3-4e1d-4cd5-9e93-cc94896c65d5">
rebased, removed the language check, and ran ruff

theomonnom reviewed (approved):
Thanks lgtm!
